actions/class.PropertiesAuthoring.php

```diff
@@ -304,12 +304,43 @@ public function getClassForm(core_kernel_classes_Class $class): tao_helpers_form
                 //save all properties values
                 if (isset($data['properties'])) {
                     $this->saveProperties($data);
+                    $this->populateSubmittedProperties($myForm, $data);
                 }
             }
         }
         return $myForm;
     }
 
+    private function populateSubmittedProperties($myForm, $data): void
+    {
+        $elementRangeArray = [];
+        $groups = $myForm->getGroups();
+        if (isset($data['properties'])) {
+            foreach ($data['properties'] as $prop) {
+                if (!empty($prop['range'])) {
+                    $elementUri = $groups['property_' . $prop['uri']]['elements'][0];
+                    if (isset($elementUri)) {
+                        $index = strstr($elementUri, '_', true);
+                        $elementRangeArray[$index . '_range_list'] = $prop['range'];
+                    }
+                }
+            }
+        }
```

Comment:
You can avoid processment

```suggestion
        if (empty($data['properties'])) {
            return;
        }
        
        foreach ($data['properties'] as $prop) {
            if (!empty($prop['range'])) {
                $elementUri = $groups['property_' . $prop['uri']]['elements'][0];
                if (isset($elementUri)) {
                    $index = strstr($elementUri, '_', true);
                    $elementRangeArray[$index . '_range_list'] = $prop['range'];
                }
            }
         }
```

Reply:
You are correct. I will change this

Reply:
Done